seatunnel-connectors-v2/connector-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/serde/BsonToRowDataConverters.java

```diff
@@ -353,15 +353,16 @@ private static boolean convertToBoolean(BsonValue bsonValue) {
     }
 
     private static double convertToDouble(BsonValue bsonValue) {
-        if (bsonValue.isDouble()) {
+        try {
             return bsonValue.asNumber().doubleValue();
+        } catch (RuntimeException e) {
+            throw new MongodbConnectorException(
+                    UNSUPPORTED_DATA_TYPE,
+                    "Unable to convert to double from unexpected value '"
+                            + bsonValue
+                            + "' of type "
+                            + bsonValue.getBsonType());
```

Comment:
How about change like this? It should be faster than try catch block.

```suggestion
        if (bsonValue.isDouble() || bsonValue.isInt32() || bsonValue.isInt64() || bsonValue.isDecimal128()) {
            return bsonValue.asNumber().doubleValue();
        }
        throw new MongodbConnectorException(
                UNSUPPORTED_DATA_TYPE,
                "Unable to convert to double from unexpected value '"
                        + bsonValue
                        + "' of type "
                        + bsonValue.getBsonType());
```

Reply:
That's great!

But the `asNumber()` method throws a exception  if the type is not one of int32, int64 and double.

https://github.com/mongodb/mongo-java-driver/blob/525c465df46f791684fccef0a1ba49f72cbba026/bson/src/main/org/bson/BsonValue.java#L79-L85

What are you think about below code?
```java
if (bsonValue.isNumber()) {
  return bsonValue.asNumber().doubleValue();
}
throw new MongodbConnectorException(
  UNSUPPORTED_DATA_TYPE,
)
```

Reply:
> What are you think about that use to the isNumber() method instead of isDouble(), isInt32() and isInt64().

Yes, we should use `isNumber()`.